src/swarm/neo/connection/ConnectionBase.d

```diff
@@ -1034,6 +1048,23 @@ abstract class ConnectionBase: ISelectClient
 
     /***************************************************************************
 
+        Requests to send outgoing data buffered by the OS for the socket
+        immediately. By default outgoing data are buffered to the maximum
+        payload of a TCP frame using the Linux TCP_CORK socket option.
+
+        This method is intended to be called right after `getPayloadForSending`
```

Comment:
We thought about a `bool` return value for `getPayloadForSending` to request flushing. This is problematic because[`RequestOnConnBase.getPayloadForSending`](https://github.com/sociomantic-tsunami/swarm/blob/v3.x.x/src/swarm/neo/connection/RequestOnConnBase.d#L1394) resumes the request handler fiber before it returns. So first the request moves on and does its business, and only when it suspends the next time `getPayloadForSending` would return, possibly significantly later.

Another idea is a `bool flush` parameter in the `send` delegate. This would work, but the parameter would need to be passed to all `RequestOnConnBase.send*` methods and stored in a `RequestOnConnBase` class variable so that `getPayloadForSending` can pass it to the `send` delegate. Considering the existing convolution of the `send/Receive/andHandleEvents)` methods this is practicable only with the redesigned one-for-all method.